aws/resource_aws_elasticsearch_domain.go

```diff
@@ -717,6 +777,10 @@ func resourceAwsElasticSearchDomainUpdate(d *schema.ResourceData, meta interface
 		input.AdvancedOptions = stringMapToPointers(d.Get("advanced_options").(map[string]interface{}))
 	}
 
+	if d.HasChange("advanced_security_options") {
+		input.AdvancedSecurityOptions = expandAdvancedSecurityOptions(d.Get("advanced_security_options").([]interface{}))
+	}
+
```

Comment:
let's remove this logic since the API doesn't allow modifying this field and will error with `Error: ValidationException: You can't modify advanced security options after creating the domain.` if you attempt to do so

Reply:
@anGie44 Even though you cannot change whether advanced security options is enabled or disabled after creating the domain, you are able to change some of the other settings such as "internal_user_database_enabled", which then requires you to change the "master_user_options" settings (master_user_name/password vs. master_user_arn). You can also change the "master_user_password" option after creating the domain, assuming that you keep "internal_user_database_enabled" set to true. When I tested this, I found that aws does actually change the password of the master user. I had a hard time finding anything about this behavior in the aws docs unfortunately.

Reply:
Should we just be checking for changes to "internal_user_database_enabled" and "master_user_options" instead here?

Reply:
@jon-fearer understood! I just confirmed on my end that `master_user_options` can indeed be updated as you've suggested. My only final note then is to have these caveats documented in the website/docs where applicable.